[Doc] Impove `scrollToTop` in buttons doc and document `_scrollToTop` in `useRedirect` (#10449)
marmelab/react-admin: docs/Buttons.md

It also supports [all the other `<Button>` props](#button).

### `scrollToTop`

> By default, react-admin scrolls to top on each redirection. You can disable it as follows:

Reviewer: ```suggestion
By default, react-admin scrolls when redirecting to the create view. You can disable it as follows:
```

> By default, react-admin scrolls when redirecting to the create view. You can disable it as follows:

Reviewer 2: ```suggestion
By default, react-admin scrolls the page to the top after redirecting to the create view. You can disable it as follows:
```

> By default, react-admin scrolls the page to the top after redirecting to the create view. You can disable it as follows:

Reviewer: ```suggestion
By default, `<CloneButton>` scrolls the page to the top after redirecting to the create view. You can disable it as follows:
```

**Tip:** To allow users to create a record without leaving the current view, use the [`<CreateInDialogButton>`](./CreateInDialogButton.md) component.

### `scrollToTop`

> By default, react-admin scrolls to top on each redirection. You can disable it as follows:

Reviewer: ```suggestion
By default, react-admin will scroll the page to the top after redirecting. You can disable it as follows:
```

> By default, react-admin will scroll the page to the top after redirecting. You can disable it as follows:

Reviewer 2: ```suggestion
By default, `<CreateButton>` scrolls the page to the top after redirecting. You can disable it as follows:
```

Same for the others

**Tip:** To allow users to edit a record without leaving the current view, use the [`<EditInDialogButton>`](./EditInDialogButton.md) component.

### `scrollToTop`

> By default, react-admin scrolls to top on each redirection. You can disable it as follows:

Reviewer: ```suggestion
By default, react-admin will scroll the page to the top after redirecting. You can disable it as follows:
```

**Tip**: If you want to link to the Show view manually, use the `/{resource}/{record.id}/show` location.

### `scrollToTop`

> By default, react-admin scrolls to top on each redirection. You can disable it as follows:

Reviewer: ```suggestion
By default, react-admin will scroll the page to the top after redirecting. You can disable it as follows:
```
